examples/bosd-payload-with-rust-bitcoin.rs

```diff
@@ -0,0 +1,108 @@
+//! Implements an example PSBT workflow.
+//!
+//! The workflow we simulate is that of a setup using a BOSD payload to create a PSBT with a P2WPKH output and a OP_return output.
+//!
+//! ## How to check:
+//!
+//! 1. Start Bitcoin Core in Regtest mode, for example:
+//!
+//!    `bitcoind -regtest -server -daemon -fallbackfee=0.0002 -rpcuser=admin -rpcpassword=pass -rpcallowip=127.0.0.1/0 -rpcbind=127.0.0.1 -blockfilterindex=1 -peerblockfilters=1`
+//!
+//! 2. Define a shell alias to `bitcoin-cli`, for example:
+//!
+//!    `alias bt="bitcoin-cli -rpcuser=admin -rpcpassword=pass -rpcport=18443"`
+//!
+//! 3. Run this example to get the Base64 encoded PSBT with:
+//!
+//!    `cargo run --features examples --example payload_with_rust-bitcoin`
+//!
+//! 4. Now that you have with the Base64 enconded PSBT:
+//!
+//!    `bt analyzepsbt cHNidP8BAEACAAAAAAIAAAAAAAAAABYAFJf1yVh7sF47WTYWjYsqsoGUbuOsAAAAAAAAAAAOagxCaXRjb2luIEJPU0QAAAAAAAAA`
+//!
+
+#[cfg(feature = "examples-dep")]
+use base64::{engine::general_purpose::STANDARD, Engine};
+
+use bitcoin::{
+    absolute::LockTime,
+    bip32::{IntoDerivationPath, Xpriv},
+    key::Secp256k1,
+    secp256k1::All,
+    transaction::Version,
+    Amount, Network, Psbt, Transaction, TxOut,
+};
+
+fn main() {
+    // This example shows the basic usage of the [`bitcoin_bosd::Descriptor`] struct
+    // which is responsible for parsing and transmuting BOSD payloads into Bitcoin [`ScriptBuf`] and [`Address`].
```

Comment:
```suggestion
    // This example shows the basic usage of the BOSD `Descriptor`
    // for parsing and transmuting into Bitcoin `ScriptBuf` and `Address`.
```

This is a comment, doclinking will not avail you, just hamper.